packages/ui/src/theme/createTheme.ts

```diff
@@ -78,7 +78,7 @@ export function createTheme(
   // flattenProperties is another internal Style Dictionary function
   // that creates an array of all tokens.
   let cssText =
-    `[data-amplify-theme="${name}"] {\n` +
+    `:root, [data-amplify-theme="${name}"] {\n` +
```

Comment:
If we define this at the root, shouldn't we _only_ define it at the root (potentially removing `[data-amplify-theme="${name}"]` ? It seems like having it defined at both locations will cause issues... Thoughts, @dbanksdesign.

Reply:
CSS variables are inherited, so in the CSSOM the :root set of variables will get overridden by the [data-amplify-theme] at that point. Since both sets are the same it shouldn't really do anything... unless someone overrode a variable at :root it would get overridden. If we do remove the [data-amplify-theme] completely we need to rethink how we handle the theme overrides, because right now those happen on that element...

Reply:
Keeping [data-amplify-theme] selector would make sure overrides like dark mode work, however because the original issue (themes not working on Portal elements like Menu), overrides would not work those elements...

Reply:
We could change how we do overrides by injecting data attributes on the html element directly...

Reply:
Yeah, I was thinking we would do all overrides at the html element level so that we can affect Portal elements.

Reply:
@jacoblogan does that make sense? We'll need to update the AmplifyProvider code for that